Formula/libosmium.rb

```diff
@@ -0,0 +1,21 @@
+class Libosmium < Formula
+  desc "Fast and flexible C++ library for working with OpenStreetMap data."
+  homepage "http://osmcode.org/libosmium/"
+  url "https://github.com/osmcode/libosmium/archive/v2.9.0.tar.gz"
+  sha256 "d4312c0cd9cca41fbf2927aee67cbb95d566540a76bb01fc643e8ba76db13419"
+
+  depends_on "cmake" => :build
+  depends_on "boost" => :build
+  depends_on "google-sparsehash" => :optional
+  depends_on "expat" => :optional
+  depends_on "gdal" => :optional
+  depends_on "proj" => :optional
+  depends_on "doxygen" => :optional
```

Comment:
How common are these? We usually try to provide sane defaults and limit the number of options.